src/proof/lean/lean_printer.h

```diff
@@ -19,31 +19,78 @@
 
 #include <iostream>
 
+#include "expr/node_algorithm.h"
+#include "expr/proof_checker.h"
 #include "expr/proof_node.h"
+#include "proof/lean/lean_rules.h"
 
 namespace CVC4 {
+
 namespace proof {
 
-enum class LeanRule : uint32_t
-{
-  // in what format should I put these lean rules
-  // all the lean rules
-};
+/**
+ * Converts a lean rule to a string.
+ * @param id The lfsc rule
+ * @return The name of the lfsc rule
+ */
+const char* toString(LeanRule id);
+
+/**
+ * Writes a lean rule name to a stream.
+ *
+ * @param out The stream to write to
+ * @param id The lean rule to write to the stream
+ * @return The stream
+ */
+std::ostream& operator<<(std::ostream& out, LeanRule id);
 
-static void leanPrinter(std::ostream& out, std::shared_ptr<ProofNode> pfn)
+class LeanPrinter
 {
-  // should print preamble
-  // should print sorts
-  // should print terms, letified
-
-  // should print theorem statement
-  out << "hello world"
-      << "\n";
-
-  // should traverse proof node, print each as a proof step, according to the
-  // LEAN_RULE id in the LeanRule enum
-  out << pfn;
-}
+ public:
+  LeanPrinter();
+  ~LeanPrinter() {}
+
+  /**
+   * Print the full proof of assertions => false by pfn.
+   */
+  static void print(std::ostream& out,
+                    const std::vector<Node>& assertions,
+                    std::shared_ptr<ProofNode> pfn);
+
+ private:
+  /**
+   * Convert a CVC4 Node holding an id to the corresponding LeanRule
+   */
+  static LeanRule getLeanRule(Node n);
+  //static Node getId(std::shared_ptr<ProofNode> n);
+  //static Node getConclusion(std::shared_ptr<ProofNode> n);
+  static void printKind(std::ostream& s, Kind k);
+  /**
+   * Convert a node to a Lean term -- must start with mk_ and take children as args
+   * Example: kind::AND (kind::EQUAL a b) c --> mkAnd (mkEq a b) c
+   */
+  static void printLeanString(std::ostream& s, Node n);
+  /**
+   * Convert from node to lean type syntax
```

Comment:
Use "Lean" rather than "lean" everywhere. Clarify why you talk about converting a type but receives a Node as input.